pkg/cmd/unpack/unpack.go

```diff
@@ -137,9 +171,16 @@ func runUnpackRecursive(ctx context.Context, opts *unpackOptions, visitedRefs []
 				docsIdx += 1
 				continue
 			}
-			_, relPath, err = filesystem.VerifySubpath(opts.unpackDir, docsEntry.Path)
-			if err != nil {
-				return fmt.Errorf("error resolving path %s for docs: %w", docsEntry.Path, err)
+			if docsEntry.LayerInfo != nil {
+				if docsEntry.LayerInfo.Digest != layerDesc.Digest.String() {
+					return fmt.Errorf("digest in config and manifest do not match in docs layer")
+				}
+				relPath = ""
+			} else {
+				_, relPath, err = filesystem.VerifySubpath(opts.unpackDir, docsEntry.Path)
+				if err != nil {
+					return fmt.Errorf("error resolving path %s for docs: %w", docsEntry.Path, err)
+				}
```

Comment:
Duplicated code for handling every layer. Should it be refactored ?

Reply:
This is just kind of the way things end up happening in Golang. I can try to rework it but in my experience it ends up decreasing readability :)

Reply:
Latest commit should make this a little clearer.